### DISCUSSION TITLE
com registration tool doesn't work as expected in win11

### DISCUSSION THREAD

**Discussion by smeveriss** [nvaccess/nvda#17495] [General]
Issue refference
https://github.com/nvaccess/nvda/issues/9039
I have a windows 11 lenovo laptop and removed office 365 trial.
This caused the need to run com reg fix tool but running it and rebooting several times didn't fix the issue.
No problem I thought I know what the issue was so firstly I decided before running the tool I would put office back in as I was doing it, the version of office was 2021 the latest at the time I understand that 2024 is the latest now but back when I did this 5 months ago that wasn't the case.
Reinstalling office should have fixed it without com reg its done that in the past.
If that doesn't work repairing adobe reader usually fixes that, it didn't.
Running com reg tool didn't fix it.
On another system of the similar configuration after removing 365, I ran com registration tool first.
This again didn't fix it, however after repairing adobe reader which didn't fix it I then reinstalled office 2021 and then it appeared functional again.
Both these systems are ones I maintain for others and luckily windows 11 has a newer interface so only ccleaner and a few older programs are spaciffically effected however I expected com registration to work.
On that note if its known that uninstalling office can cause deregistration of some important libraries or even removal of them shouldn't there be a microsoft fixit?
Now I haven't found one but of course microsoft may not call the fixit com registration.
I'm also not sure how to test this as I have my own hp box on win10 and I have no wish to screw it up.
However as I move to upgrade my main lenovo business network desktop with all the pictures and other things I'd really like a solution I could rely on.
I understand the com reg tool could be relied on.
Now when it previously happened on my main box I just reinstalled from scratch and since I rarely remove office I never encounter this drama but obviously its impractical to well reload from scratch especially if it is another person's system especially if I am effected but don't need to reuse the system.
Interestingly if I remove all settings of nvda and reinstall it the system acts completely normal to a point but when I reboot it just doesn't work again.
Now as I write this it occurs to me I should have pulled the error log but I don't have access to the system just now and may not for some time and never thought about outputting a log amd bringing it with me so this will have to be a general issue/ question rather than spaciffics so I am fine if I get a generalised answer and not machine spaciffic.

**Comment by Adriani90** [comment]
Did you install a very old version of Adobe Reader or MS Office before installing the Office 365 for the first time? Probably there is still some really old stuff in your program data which produces this COM DLL issues.

You could try running these commands in cmd.exe executed with administrator rights.

```
Dism.exe /online /Cleanup-Image /StartComponentCleanup /ResetBase
```

This will basically remove all older versions of DLL files from old software verisons that you might have installed before.

```
DISM /online /cleanup-image /scanhealth
```

This will scan your windows image for any damages

```
DISM /Online /Cleanup-Image /CheckHealth
```

This will run a check whether damages have been found after the scan. It will promt you whether your Windows image is damaged and whether it can be repaired or not.
If there are damages and they can be repaired, you can run:

```
DISM /online /cleanup-image /restorehealth 
```

to repair the image.

With
```
Dism /Cleanup-Mountpoints
```

you can repair the Windows image afterwards in case it is still having broken DLL files.

Finally, run

```
SFC /SCANNOW
```

This will search for damaged system files and will repair them automatically.

Afterwards, restart the computer. In most cases the issues should be fixed.
